libs/authentication/src/lib/components/login/login.component.ts

```diff
@@ -1,15 +1,50 @@
-import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { NgForm } from '@angular/forms';
+import { AuthService } from '../../services/auth.service';
+import { first } from 'rxjs/operators';
+import { Router } from '@angular/router';
+import { CookieService } from 'ngx-cookie-service';
 
 @Component({
   selector: 'container-management-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class LoginComponent implements OnInit {
-  hide: boolean = true;
+export class LoginComponent {
+  hide = true;
+  errorMessage = '';
 
-  constructor() {}
+  constructor(
+    private readonly authService: AuthService,
+    private readonly router: Router,
+    private readonly cookieService: CookieService
+  ) {}
 
-  ngOnInit(): void {}
+  onSubmit(loginForm: NgForm) {
+    if (!loginForm.valid) {
+      this.errorMessage = 'Please enter valid username and password!';
+    } else {
+      this.authService
+        .login({
+          userName: loginForm.value.userName,
+          password: loginForm.value.password,
+        })
+        .pipe(first())
+        .subscribe(
+          (response) => {
+            this.errorMessage = '';
+            this.cookieService.set('token', response.token, 1);
+            localStorage.setItem('userName', loginForm.value.userName);
+            localStorage.setItem('imageMaxSizes', response.imageMaxSizes);
+            localStorage.setItem('serverSettings', response.settings);
+            //this.adjustUploadSettings(response);
+            this.router.navigate(['/', 'camera']);
```

Comment:
Please help add this logic after login successfully
```
adjustUploadSettings(response) {
      localStorage.setItem('uploadSettings', JSON.stringify(this.uploadSettings));

      if (localStorage.getItem('uploadSettings')) {
        const uploadSettings = JSON.parse(localStorage.getItem('uploadSettings'));
        const serverSettings = JSON.parse(response.data.settings);
        uploadSettings.ftp.enabled = uploadSettings.ftp.enabled && serverSettings.ftp.enabled;
        uploadSettings.cloudinary.enabled = uploadSettings.cloudinary.enabled && serverSettings.cloudinary.enabled;
        uploadSettings.local.enabled = serverSettings.local.enabledHigh || serverSettings.local.enabledLow;
        localStorage.setItem('uploadSettings', JSON.stringify(uploadSettings));
      }
    },
```

Reply:
Thank you for your comment.
I will handle this logic in another component.